src/engine/DataObjects.hpp

```diff
@@ -390,6 +390,17 @@ class Color : public Vector4 {
 public:
     Color();
     Color(float r, float g, float b, float a);
+
+    /// @brief Constructs a Color object from a Vector4. This assumes that the
+    /// Vector4 contains values from 0.0-1.0.
+    /// @param vec A vector containing the data to put in the Color.
+    Color(const Vector4 &vec) : Vector4(vec) {
+        SetR(vec.x());
+        SetG(vec.y());
+        SetB(vec.z());
+        SetA(vec.w());
+    }
```

Comment:
This will assign the rgba values twice. The Vector3 constructor already assigns the values.

Reply:
That's very true.